src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -52,13 +52,38 @@ import { attachInputBoxStyler } from 'vs/platform/theme/common/styler';
 import { IThemeService } from 'vs/platform/theme/common/themeService';
 
 export class FileDataSource implements IDataSource {
+	private toDispose: IDisposable[] = [];
+	private enableVirtualDirectories: boolean;
+	private virtualDirectoryPatterns: { [glob: string]: string[] };
```

Comment:
Is that right place to do it? Or in other words: Is the `FileDataSource` the only data source that supports Nesting?